spacewalk/uyuni-setup-reportdb/bin/uyuni-setup-reportdb

```diff
@@ -0,0 +1,436 @@
+    [ -z "$ADDRESS" ] && ADDRESS="*"
+    [ -z "$REMOTE" ] && ask_check true "Remote addresses to allow connection from (address/netmask format, comma-separated): " REMOTE "^$Remote_RE\$"
+
+    postgresql_service enable
+
+    if [ ! -d "$PG_DATA/base" ]; then
+        PGHOME=$(getent passwd postgres | awk -F: '{print $6}')
+        if /usr/sbin/selinuxenabled; then
+            SEMODE=$(/usr/sbin/getenforce)
+            if [ "$SEMODE" = 'Permissive' -o "$SEMODE" = 'Enforcing' ]; then
+                RCONOUT=$(/sbin/restorecon -Rnv $PGHOME)
+                if [ -n "$RCONOUT" ]; then
+                    echo
+                    echo "The directory \"$PGHOME\" does not seem to have correct SELinux context."
+                    echo
+                    echo "Please fix the SELinux context by running:"
+                    echo
+                    echo "    restorecon -Rv $PGHOME"
+                    echo
+                    exit 1
+                fi
+            fi
+        fi
+        echo -e 'LC_CTYPE=en_US.UTF-8\nexport LC_CTYPE' >$PGHOME/.i18n
+        postgresql_service initdb
+    fi
+
+    if $STANDALONE; then
+        sed -i 's/^\(\s*listen_addresses.*\)$/### next line has been commented out by uyuni-setup-reportdb ###\n##\1/' $POSTGRESQL
+        sed -i 's/^\(\s*max_connections.*\|\s*shared_buffers.*\)$/### next line has been commented out by uyuni-setup-reportdb ###\n##\1/' $POSTGRESQL
+
+        cat >> $POSTGRESQL <<EOF
+### uyuni-setup-reportdb modified values
+checkpoint_completion_target = 0.7
+effective_cache_size = 1152MB
+log_line_prefix = '%m '
+maintenance_work_mem = 96MB
+max_connections = 100
+shared_buffers = 384MB
+wal_buffers = 4MB
+work_mem = 2560kB
+EOF
+    fi
+
+    # SSL setup is required when opening DB to the world
+    postgres_reconfig "ssl" "on"
+    postgres_reconfig "ssl_cert_file" "'$SSL_CERT'"
+    postgres_reconfig "ssl_key_file" "'$SSL_KEY'"
+
+    echo "$ADDRESS"|grep -q '127.0.0.1\|*' || ADDRESS="$ADDRESS, 127.0.0.1"
+    postgres_reconfig "listen_addresses" "'$ADDRESS'"
+
+    sysctl kernel.shmmax | ( read v v v ; LIMIT=5000000000 ; if (("$v"0 < $LIMIT )) ; then sysctl "kernel.shmmax=$LIMIT" >> /etc/sysctl.conf ; fi )
+
+    if $STANDALONE; then
+        sed -i 's/^\([^#].*\)$/### next line has been commented out by uyuni-setup-reportdb: ###\n##\1/ ' $PG_HBA
+    fi
+    cat >> $PG_HBA <<EOF
+local $PGNAME $PGUSER scram-sha-256
+host  $PGNAME $PGUSER 127.0.0.1/8 scram-sha-256
+host  $PGNAME $PGUSER ::1/128 scram-sha-256
+EOF
+
+    for ADDR in $(echo $REMOTE|sed 's/,/ /g'); do
+        cat >> $PG_HBA <<EOF
+host $PGNAME $PGUSER $ADDR scram-sha-256
+EOF
+    done
+
+    # Create & configure /etc/rhn/rhn.conf
+    RHN_CONF_DIR=$(dirname $RHN_CONF)
+    if [ -d $RHN_CONF_DIR ]; then
+        chmod 0710 $RHN_CONF_DIR
+    else
+        mkdir -m 0710 $RHN_CONF_DIR
+    fi
+
+    if $STANDALONE; then
+        cat > $RHN_CONF <<EOF
+report_db_backend = postgresql
```

Comment:
What about using "reporting.db_backend", "reporting.db_user" and so on so we take advantage of the existing namespace functionality? Could we maybe use a separate rhn_reporting.conf for all reporting related configuration?

Reply:
Ah no we can't :man_facepalming: I tested this more deeply and `db_user` in `rhn.conf` actually overrides `reporting.db_user` so we strictly need to change it.